test/tests/functions/number.json

```diff
@@ -0,0 +1,409 @@
+      ],
+      "exp": "one other"
+    },
+    {
+      "src": ".match {$foo :number} {$bar :number} one one {{one one}} one * {{one other}} * * {{other}}",
+      "params": [
+        {
+          "name": "foo",
+          "value": 2
+        },
+        {
+          "name": "bar",
+          "value": 2
+        }
+      ],
+      "exp": "other"
+    },
+    {
+      "src": ".input {$foo :number} .match {$foo} one {{one}} * {{other}}",
+      "params": [
+        {
+          "name": "foo",
+          "value": 1
+        }
+      ],
+      "exp": "one"
+    },
+    {
+      "src": ".local $foo = {$bar :number} .match {$foo} one {{one}} * {{other}}",
+      "params": [
+        {
+          "name": "bar",
+          "value": 1
+        }
+      ],
+      "exp": "one"
+    },
+    {
+      "src": ".input {$foo :number} .local $bar = {$foo} .match {$bar} one {{one}} * {{other}}",
+      "params": [
+        {
+          "name": "foo",
+          "value": 1
+        }
+      ],
+      "exp": "one"
+    },
+    {
+      "src": ".input {$bar :number} .match {$bar} one {{one}} * {{other}}",
+      "params": [
+        {
+          "name": "bar",
+          "value": 2
+        }
+      ],
+      "exp": "other"
+    },
+    {
+      "src": ".input {$bar} .match {$bar :number} one {{one}} * {{other}}",
+      "params": [
+        {
+          "name": "bar",
+          "value": 1
+        }
+      ],
+      "exp": "one"
+    },
+    {
+      "src": ".input {$bar} .match {$bar :number} one {{one}} * {{other}}",
+      "params": [
+        {
+          "name": "bar",
+          "value": 2
+        }
+      ],
+      "exp": "other"
+    },
+    {
+      "src": ".input {$bar} .match {$bar :number} one {{one}} * {{other}}",
+      "params": [
+        {
+          "name": "bar",
+          "value": 1
+        }
+      ],
+      "exp": "one"
+    },
+    {
+      "src": ".input {$bar} .match {$bar :number} one {{one}} * {{other}}",
+      "params": [
+        {
+          "name": "bar",
+          "value": 2
+        }
+      ],
+      "exp": "other"
+    },
+    {
+      "src": ".input {$none} .match {$foo :number} one {{one}} * {{{$none}}}",
+      "params": [
+        {
+          "name": "foo",
+          "value": 1
+        }
+      ],
+      "exp": "one"
+    },
+    {
+      "src": ".local $bar = {$none} .match {$foo :number} one {{one}} * {{{$bar}}}",
+      "params": [
+        {
+          "name": "foo",
+          "value": 1
+        }
+      ],
+      "exp": "one"
+    },
+    {
+      "src": ".local $bar = {$none} .match {$foo :number} one {{one}} * {{{$bar}}}",
+      "params": [
+        {
+          "name": "foo",
+          "value": 2
+        }
+      ],
+      "exp": "{$none}",
+      "expErrors": [
+        {
+          "type": "unresolved-variable"
+        }
+      ]
+    },
+    {
+      "src": "{42 :number @foo @bar=13}",
+      "exp": "42",
+      "expParts": [
+        {
+          "type": "literal",
+          "value": "42"
```

Comment:
The original expParts was:
`[{ "type": "number", "parts": [{ "type": "integer", "value": "42" }] }]`
Since the part doesn't have the `source` property that a message _expression_ part requires, I've assumed we want this as a string literal part.